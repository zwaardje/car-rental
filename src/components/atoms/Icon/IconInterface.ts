import { HTMLAttributes } from 'react';

export type IconTypes =
'add' |
'back'|
'calendar' |
'call' |
'cancel' |
'check' |
'clock' |
'delete' |
'edit' |
'info' |
'lightning' |
'location' |
'lock' |
'luggage' |
'person' |
'plane' |
'search' |
'star' |
'tie' |
'leaf' |
'addressbook' |
'car' |
'fullscreen' |
'car-location' |
undefined;

export interface Icon extends HTMLAttributes<any> {
    icon: IconTypes | undefined;
}
