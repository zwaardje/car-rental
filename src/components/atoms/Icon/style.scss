.icon {
    width: 16px;
    height: 16px;
    min-width: 12px;
    min-height: 12px;
}
