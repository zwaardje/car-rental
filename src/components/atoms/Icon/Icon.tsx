import * as React from 'react';

import { Icon as IconInterface } from './IconInterface';

import './style.scss';

export interface Props extends IconInterface {}
export interface State {}

export default class Icon extends React.Component<Props, State> {
    static displayName = 'Icon';

    defaultIconUrl: string = '/static/icons/iconset.svg';

    get iconRef(): string {
        const { icon } = this.props;
        const url =  `${ this.defaultIconUrl }#icon-${ icon }`;

        return url;
    }

    render() {
        const { icon, className } = this.props;

        if (!icon) return null;

        return (
            <svg className={`icon icon-${icon}${className ? ` ${className}` : ''}`}>
                <use xlinkHref={this.iconRef} href={this.iconRef}></use>
            </svg>
        );
    }
}
