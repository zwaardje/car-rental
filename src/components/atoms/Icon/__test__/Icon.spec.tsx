import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Icon from '../Icon';

import {
    check,
} from '../__data__/Icon.mockdata';

describe('Icon', () => {

    it('renders check icon correctly', () => {
        const props = { ... check };
        const tree = renderer.create(<Icon {... props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
