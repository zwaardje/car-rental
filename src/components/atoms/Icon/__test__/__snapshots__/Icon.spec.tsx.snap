// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders check icon correctly 1`] = `
<svg
  className="icon icon-location"
>
  <use
    href="/assets/icons/iconset.svg#icon-location"
    xlinkHref="/assets/icons/iconset.svg#icon-location"
  />
</svg>
`;
